lib/manageiq_foreman/lib/manageiq_foreman/connection.rb

```diff
@@ -17,6 +17,11 @@ def initialize(connection_attrs)
       @connection_attrs = connection_attrs
     end
 
+    def verify?
+      results = Array(fetch(:home).try(:results)).first
+      results.respond_to?(:key?) && results.key?("links")
```

Comment:
Oh, results could actually be real data that's invalid, not necessarily nil.

For example, if you point ManageiqForeman at "google.com", you actually get something back, but it's not the valid parsed JSON that's expected...it's just a big string.